src/grt/src/GlobalRouter.cpp

```diff
@@ -620,13 +622,84 @@ void GlobalRouter::updateDirtyNets(std::vector<Net*>& dirty_nets)
         logger_->error(
             GRT, 267, "Net {} has disconnected segments.", net->getName());
       }
+    } else if (net->getPins().size() >= 2) {
```

Comment:
I think you'll need to check the function `newPinOnGrid` in the first if statement too. The way it is now, your code is probably never being called.

Reply:
You'll also need to check it again in this if. So your clause will be something like `else if (!newPinOnGrid() && net->getPins().size() >= 2)`